standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5213,11 +5213,13 @@ public LockHandle acquireLock(String key) throws MetaException {
     Connection dbConn = null;
     Statement stmt = null;
     ResultSet rs = null;
+    boolean needToCloseConn = false;
     try {
       try {
         String sqlStmt = sqlGenerator.addForUpdateClause("SELECT \"MT_COMMENT\" FROM \"AUX_TABLE\" WHERE \"MT_KEY1\"=" + quoteString(key) + " and \"MT_KEY2\"=0");
         lockInternal();
         dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED, connPoolMutex);
+        needToCloseConn = true;
```

Comment:
that won't work if thread was interrupted right after getting the connection, set it to true by default.

Reply:
Also I think this LockHandle should implement AutoCloseable and close active connection if any.

Reply:
AutoCloseable may not be necessary, lockHandle implement releaseLocks()  which close the connection.  It uses with acquireLock in our codes in following pattern:
   TxnStore.MutexAPI.LockHandle handle = null;
    try {
      handle = txnHandler.getMutexAPI().acquireLock(TxnStore.MUTEX_KEY.TxnCleaner.name());
      long start = System.currentTimeMillis();
      txnHandler.cleanEmptyAbortedAndCommittedTxns();
      LOG.debug("Txn cleaner service took: {} seconds.", elapsedSince(start));
    } catch (Throwable t) {
      LOG.error("Unexpected error in thread: {}, message: {}", Thread.currentThread().getName(), t.getMessage(), t);
    } finally {
      if (handle != null) {
        handle.releaseLocks();
      }
    }

Reply:
hopefully it's done like this everywhere, otherwise we might get same issue but in different place

Reply:
Yes, I checked every place that acquireLock is called, they are always matching.